.travis.yml

```diff
@@ -4,32 +4,37 @@ services:
   - docker
 script:
   - echo FROM ocaml/opam:${DISTRO}_ocaml-${OCAML_VERSION} > Dockerfile.test
-  - echo VOLUME /bd    >> Dockerfile.test
-  - echo VOLUME /repo  >> Dockerfile.test
-  - echo WORKDIR /repo >> Dockerfile.test
-  - echo "BASE_REMOTE=${BASE_REMOTE}"      > env.list
-  - echo "EXTRA_REMOTES=${EXTRA_REMOTES}" >> env.list
-  - echo "COMPILE_ALL=${COMPILE_ALL}"     >> env.list
-  - echo "OPAM_LINT=${OPAM_LINT}"         >> env.list
-  - echo "OCAML_VERSION=${OCAML_VERSION}" >> env.list
+  - echo COPY . /xs-opam >> Dockerfile.test
+  - echo "ENV BASE_REMOTE=${BASE_REMOTE}"     >> Dockerfile.test
+  - echo "ENV EXTRA_REMOTES=${EXTRA_REMOTES}" >> Dockerfile.test
+  - echo "ENV COMPILE_ALL=${COMPILE_ALL}"     >> Dockerfile.test
+  - echo "ENV OPAM_LINT=${OPAM_LINT}"         >> Dockerfile.test
+  - echo "ENV OCAML_VERSION=${OCAML_VERSION}" >> Dockerfile.test
+  - echo WORKDIR /xs-opam                     >> Dockerfile.test
+  - echo RUN bash travis.sh                   >> Dockerfile.test
   - docker build -t local-build -f Dockerfile.test .
-  - docker run --env-file=env.list -v ~/build/$TRAVIS_REPO_SLUG:/repo -v $TRAVIS_BUILD_DIR:/bd local-build bash travis.sh
+after_success:
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" -a "$TRAVIS_BRANCH" = "master" -a -n "$DOCKER_HUB_TAG" ]; then
```

Comment:
When we start uploading these images from multiple xs-opam branches, we'll have to change this logic a bit.